store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -8156,6 +8162,38 @@ public void modifyContact(OperationContext octxt, int contactId, ParsedContact p
         return result;
     }
 
+    /**
+     * It modifies contact's first and full name with display name
+     * @param octxt
+     * @param listOfContactsToBeModified This list consist list of Map, InternetAddress as key and Pair of ContactId and 
+     * DisplayName as Value
+     */
+    public void modifyAutoContact(OperationContext octxt, List<Map<InternetAddress, Pair<Integer, String>>> listOfContactsToBeModified)
+            throws IOException {
+        String email = null;
+        String locale = octxt != null && octxt.getAuthenticatedUser() != null ? octxt.getAuthenticatedUser().getPrefLocale() : null;
+        boolean nameFormatLastFirst = false;
+        if (locale != null && locale.equals(JAPANESE_LOCALE_CODE)) {
+            nameFormatLastFirst = true;
+        }
+        for (Map<InternetAddress, Pair<Integer, String>> map : listOfContactsToBeModified) {
+            try {
+                for (InternetAddress address : map.keySet()) {
+                    email = address.getAddress();
+                    ZimbraLog.mailbox.debug("Modifying contact addr=%s", email);
+                    ParsedContact pc = new ParsedContact(new ParsedAddress(address, nameFormatLastFirst).getAttributes());
+                    String[] splited = map.get(address).getSecond().split(" ");
+                    pc.modifyField(ContactConstants.A_firstName, splited[0]);
+                    pc.modifyField(ContactConstants.A_fullName, map.get(address).getSecond());
+                    pc.modifyField(MailConstants.A_EMAIL, email);
```

Comment:
`ParsedContact pc = new ParsedContact(new ParsedAddress(address, nameFormatLastFirst).getAttributes());` seems to parse displayName into fullName, middleName and firstName. `pc` has also an email address.
I think we can remove lines 8185-8188.

Reply:
Done, thanks!